ImageFeed/Presentation/ImagesList/ImagesListViewController.swift

```diff
@@ -31,16 +35,52 @@ final class ImagesListViewController: UIViewController {
         imagesListView.tableView.delegate = self
         
         imagesListView.tableView.rowHeight = 200
+        
+        imagesListServiceObserver = NotificationCenter.default
```

Comment:
Можно использовать вот этот метод - [addobserver](https://developer.apple.com/documentation/foundation/notificationcenter/1415360-addobserver) и не нужно будет руками удалять обсервера

Reply:
вот этот момент я бы хотел прояснить, написано, что удалять оbserver нужно и нставник говорил мне именно так сделать, уже задавал такой вопрос, на прошлом ревью также указали не необходимость удаления оbserver'a

<img width="799" alt="Screenshot 2024-08-07 at 23 23 54" src="https://github.com/user-attachments/assets/4dd53e53-ff78-46a0-b43f-c63dbb06b718">

и еще вот
https://reintech.io/blog/mastering-swift-notification-center

Reply:
Есть два метода доавления обсервера, которые довольно похожи:
1. [Раз](https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver) - ты его используешь
2. [Два](https://developer.apple.com/documentation/foundation/notificationcenter/1415360-addobserver) - я предложил использовать его

И дальше стоит посмотреть описание метода [removeObserver](https://developer.apple.com/documentation/foundation/nsnotificationcenter/1413994-removeobserver), а именно вот эти строки:

> If your app targets iOS 9.0 and later or macOS 10.11 and later, and you used [addObserver:selector:name:object:](https://developer.apple.com/documentation/foundation/nsnotificationcenter/1415360-addobserver), you do not need to unregister the observer. If you forget or are unable to remove the observer, the system cleans up the next time it would have posted to it.

Вот, поэтому для второго метода можно не вызывать removeObserver-a, поэтому я советую использовать именно его. Такие вот хитрые нюансы.

Молодец, что обратил внимание 🔥

Reply:
@Andreynnt вроде улеглось, таким образом, получается, что "старое API" лучше/удобнее "нового" с кложурами?